conans/test/functional/profiles/profiles_test.py

```diff
@@ -39,3 +39,37 @@ def test_env_var_without_variable_substitution(self):
         self.assertIn("bar=other;new value", client.out)
         self.assertIn("zet=$KK", client.out)
         print(client.out)
+
+    def test_config_parser(self):
+        from configparser import ConfigParser, BasicInterpolation
+
+        class CustomInterpolation(BasicInterpolation):
+            """Interpolation which expands environment variables in values."""
+
+            def before_get(self, parser, section, option, value, defaults):
+                rawval = parser.get(section, option, raw=True)
+                val = rawval % parser.defaults()
+                return val
+
+        parser1 = ConfigParser(interpolation=CustomInterpolation())
+        profile1 = """
+        [env]
+        bar=bar1
+        zet=zet
+        CXXFLAGS=-fpic
+        """
+        parser1.read_string(profile1)
+        p1_vars = dict(parser1.items("env"))
+        self.assertDictEqual({"bar": "bar1", "cxxflags": "-fpic", "zet": "zet"}, p1_vars)
+
+        parser2 = ConfigParser(interpolation=CustomInterpolation(), defaults=p1_vars)
+        profile2 = """
+        [env]
+        foo=foo1
+        bar=bar2
+        CXXFLAGS=%(cxxflags)s -fother
```

Comment:
Could we better use the flags with the same casing, ``%(CXXFLAGS)`` in this case?

Reply:
done